### PR TITLE
US4 Slice 1 Task 2: Add suggestNextAction and wire into scanner Phase 4

### PR DESCRIPTION
**Pull request by JMBattista**
## Summary
- **Primary outcome:** Implement User Story 4 of `smithy-status-skill` — the pure `suggestNextAction` function, its integration as scanner Phase 4 (ancestor-suppression pass), and the text-mode hint renderer. `smithy status` now surfaces deterministic next-action suggestions on every classified record, both in JSON output and as copy-pasteable hint lines in the default text tree.
- **Notable behaviour changes:**
  - `scan()` gains a fourth phase: after classification, it walks each record's `parent_path` chain (with cycle guard) to detect a `not-started` ancestor, then assigns `record.next_action` via `suggestNextAction`. Done/unknown records get `null`; read-error records keep `next_action` omitted so the absence of the field continues to mean "never evaluated".
  - `renderTree` accepts an optional `renderHints` flag; `smithy status` default text mode turns it **on** so every actionable, non-suppressed record emits an indented `→ <command> <args>` hint line. Done records and records with `suppressed_by_ancestor: true` emit no hint line. JSON output is byte-identical to Slice 1 (hints never leak into JSON).
- **Follow-up work deferred:** None for US4. US5 (skill passthrough) and US6 (filter flags) are separate stories and consume this work unchanged.

## Context
- **Why now:** US4 is the "actionable" half of the status feature. Without it, `smithy status` is a diagnostic; with it, the output is a worklist that tells the user which smithy command to run next for each in-progress or not-started artifact.
- **User story:** Developers using `smithy status --format json` see deterministic next-action data on every record; developers using the default terminal output see a copy-pasteable hint line beneath each actionable item.
- **Spec reference:** FR-010 (deterministic rule table) and FR-011 (ancestor suppression) in `specs/2026-04-12-004-smithy-status-skill/smithy-status-skill.spec.md`; AS 4.1–4.5. Resolves SD-014, SD-015, SD-016.

### Key architectural choices
1. **Pure function design:** `suggestNextAction(record, resolvedChildren, ancestorNotStarted): NextAction | null` is pure — no I/O, no mutation. It mirrors the `classifier.ts` precedent so the scanner can compose the two functions without state.
2. **Deterministic rule table (FR-010 / SD-014 / SD-015):**
   - `rfc` → `smithy.render [record.path]`
   - `features` → `smithy.mark [record.path, <first-not-started-F<N>-digits>]` (fallback `[record.path]`)
   - `spec` → `smithy.cut [<spec-folder>, <first-not-started-US<N>-digits>]` (fallback `[<spec-folder>]`). `<spec-folder>` is `dirname(record.path)` for real `.spec.md` files and the path itself (with trailing slashes stripped) for virtual spec records emitted at folder paths.
   - `tasks` → `smithy.forge [record.path]`
   - `done` / `unknown` → `null` always
3. **Ancestor suppression (FR-011):** When `ancestorNotStarted` is `true`, the returned `NextAction` includes `suppressed_by_ancestor: true`. When `false`, the field is omitted entirely (not set to `false`) so the JSON payload stays sparse and absence means "not suppressed".
4. **Scanner Phase 4:** Added after the existing Phase 3 classification loop. For each record without a `read_error` warning, the scanner walks the `parent_path` chain — terminating on `null`, `undefined`, `''`, or a seen-set cycle guard — then calls `suggestNextAction`. Records carrying a `read_error:` warning are skipped entirely so the absence of `next_action` signals "never evaluated" to JSON consumers (distinct from `null` which means "evaluated, no action").
5. **Text-mode hints (SD-016):** `formatNextAction(action)` is a pure formatter that returns `→ <command> [args…]` (or just `→ <command>` when args is empty) — exactly one line, no embedded newlines, Unicode U+2192 arrow. `renderTree` gained an optional `renderHints` option (default `false` to preserve the existing render tests). The `smithy status` text-mode branch turns it on, so every actionable, non-suppressed, non-group record gets an indented hint line beneath its tree line. `--format json` output is unaffected.

### Impacted modules
- **`src/status/suggester.ts`** (new): Pure `suggestNextAction` + `formatNextAction`. Zero I/O, exported from the barrel.
- **`src/status/scanner.ts`**: New Phase 4 pass that populates `record.next_action` on every classified record, with cycle-safe `parent_path` walk for suppression. Module JSDoc refreshed from "three phases" to "four phases".
- **`src/status/render.ts`**: `RenderTreeOptions.renderHints?: boolean`; helper `maybePushHint` emits the indented hint line directly below each non-group, non-done, non-suppressed record.
- **`src/status/index.ts`**: Re-exports `suggestNextAction` and `formatNextAction`.
- **`src/commands/status.ts`**: Text-mode branch now calls `renderTree(tree, { color, renderHints: true })`. JSON branch untouched.

### Test coverage
- `src/status/suggester.test.ts`: rule table, purity, suppression flag on/off, virtual records, fallback paths, virtual-spec-folder preservation, and formatter edge cases (empty args → `→ <command>`).
- `src/status/scanner.test.ts`: five new Phase 4 integration tests including a three-level not-started chain that exercises suppression and a read-error record whose `next_action` stays omitted.
- `src/status/render.test.ts`: `renderHints` true/false, done, suppressed, nested, and group-sentinel cases.
- `src/cli.test.ts`: end-to-end fixture with actionable + suppressed + done records; asserts hint placement in text mode and `next_action` shapes in JSON.

### Validation
- `npm test`: 458/458 passing across 15 files.
- `npm run typecheck`: clean.
- `npm run build`: clean, 83.49 KB ESM bundle.

https://claude.ai/code/session_01RNfeSynNdiVwasYaX2a9WV

### CODE REVIEW

File: src/status/suggester.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    case 'spec': {
+      command = 'smithy.cut';
+      const folder = path.dirname(record.path);
+      const digits = firstNotStartedRowDigits(record, resolvedChildren);
+      if (digits !== undefined) {
+        args = [folder, digits];
+        reason = `Spec ${record.title} has a not-started user story US${digits}; run smithy.cut to decompose it into tasks.`;
+      } else {
+        args = [folder];
+        reason = `Spec ${record.title} is ${record.status}; run smithy.cut to decompose its stories into tasks.`;
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply src/status/suggester.ts:166]: Fixed in d0b1331: `suggestNextAction` now uses `specFolderFromPath()`, which strips trailing slashes for virtual spec records (path already equals the folder, e.g. `specs/webhooks/`) and only calls `path.dirname()` on real `.spec.md` file paths. Added regression tests for both the rowless fallback and the matched-row variant of virtual spec records.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/status/suggester.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        reason = `Features map ${record.title} has a not-started feature F${digits}; run smithy.mark to produce its spec.`;
+      } else {
+        args = [record.path];
+        reason = `Features map ${record.title} is ${record.status}; run smithy.mark to produce its next spec.`;
+      }
+      break;
+    }
+    case 'spec': {
+      command = 'smithy.cut';
+      const folder = path.dirname(record.path);
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply src/status/suggester.ts]: Fixed in d0b1331 (same underlying bug as #discussion_r3098577009). `suggestNextAction` now routes `record.path` through `specFolderFromPath()` before using it as the first `smithy.cut` argument: trailing-slash paths (virtual spec records emitted by the scanner at folder paths like `specs/webhooks/`) have their trailing slashes stripped and are returned as-is; real `.spec.md` file paths continue to flow through `path.dirname()`. Regression tests cover both the rowless fallback and the matched-row case.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/commands/status.ts [hunk 3]
```diff
@@ -209,8 +210,18 @@ export function statusAction(opts: StatusOptions = {}): void {
...
+  // non-suppressed record. Done records (`next_action: null`) and
+  // suppressed records (`suppressed_by_ancestor: true`) emit no hint.
+  // The `--format json` branch above is untouched — this flag only
+  // affects text-mode output (SD-016).
   const tree = buildTree(records);
-  const rendered = renderTree(tree, { color: opts.color !== false });
+  const rendered = renderTree(tree, {
+    color: opts.color !== false,
+    renderHints: true,
+  });
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply src/commands/status.ts:272]: Resolved by updating the PR description. The code is intentional — US4 Slice 2's goal is specifically to make text-mode hints visible by default (spec: "The `smithy status` text-mode output renders a copy-pasteable next-action hint beneath every in-progress or not-started record…"; acceptance scenarios 4.1–4.5 all assert the hint appears in the default text output). The original PR description was an auto-generated artifact that inaccurately claimed the flag was gated off pending design review; I've rewritten it to match the spec-mandated behavior (hints on by default in text mode; JSON output is byte-identical to Slice 1). `renderHints` remains a `RenderTreeOptions` field (default `false`) so existing render tests keep passing and downstream callers can opt out, but `statusAction` turns it on because that is the shipped feature.

---
_Generated by [Claude Code](https://claude.ai/code)_
